### DISCUSSION TITLE
Filtering orders based on shipping method_id

### DISCUSSION THREAD

**Discussion by Sartoric** [woocommerce/woocommerce#46077] [WooCommerce HPOS]
I'm wondering if it is possible to filter the orders based on a specific itemmeta of the order_item "shipping", using the wc_get_orders() or any other non "raw SQL" way. 

eg. retrieve the orders shipped with a specific custom method
  with the pair "method_id" => "MINE" in the  order_itemmeta table  (related to the order_item "shipping") 

Any ideas ?

**Comment by barryhughes** [comment]
Unfortunately, we don't see a better path at present that would not involve writing some custom SQL.

Are you happy to share *why* you want this? We'd love to learn more about the use case.

**Comment by Sartoric** [comment]
Well, basically in a plugin that add some custom shipping methods there is the need to retrieve all the orders related to these methods for a paged list view.

It would be nice to avoid raw SQL in order to prevent the filter to break in case of schema changes
Not to mention that the code would be easier to read
